planner/core/task.go

```diff
@@ -106,14 +113,19 @@ func attachPlan2Task(p PhysicalPlan, t task) task {
 
 // finishIndexPlan means we no longer add plan to index plan, and compute the network cost for it.
 func (t *copTask) finishIndexPlan() {
-	if !t.indexPlanFinished {
-		t.cst += t.count() * netWorkFactor
-		t.indexPlanFinished = true
-		if t.tablePlan != nil {
-			t.tablePlan.(*PhysicalTableScan).stats = t.indexPlan.statsInfo()
-			t.cst += t.count() * scanFactor
-		}
+	if t.indexPlanFinished {
+		return
+	}
+	cnt := t.count()
+	t.indexPlanFinished = true
+	// Network cost of transferring rows of index scan to TiDB.
+	t.cst += cnt * netWorkFactor * t.tableStats.HistColl.GetAvgRowSize(t.indexPlan.Schema().Columns, true)
```

Comment:
For index scan of IndexLookupReader, IIRC, currently we are not just returning the handle column to TiDB, instead, we return all columns of the index.

I used to make an attempt to improve this in https://github.com/pingcap/tidb/pull/10836, but found it cannot work now because TiKV/mockTiKV needs changes as well. We can improve this later when projection can be pushed down to coprocessor.